### PR TITLE
PP-5395 Refactor get payment events

### PR DESCRIPTION
**Pull request by SandorArpa**
## WHAT YOU DID

- refactor get payment events to use `GetPaymentEventService`
- separate json deserialisation and serialisation into incoming `PaymentEvents` and `PaymentEvent` (from connector) into the outgoing ` PaymentEventsResponse` and `PaymentEventResponse`
- add pact test

### CODE REVIEW

File: src/main/java/uk/gov/pay/api/model/PaymentEvent.java [hunk 1]
```diff
@@ -1,64 +1,39 @@
...
-import io.swagger.annotations.ApiModelProperty;
-import uk.gov.pay.api.model.links.PaymentEventLink;
 
-@ApiModel(value="PaymentEvent", description = "A List of Payment Events information")
+@JsonIgnoreProperties(ignoreUnknown = true)
 public class PaymentEvent {
-    @JsonProperty("payment_id")
-    private final String paymentId;
+    @JsonProperty("charge_id")
+    private String chargeId;
```
Comment (jankowiakmaria) [inline src/main/java/uk/gov/pay/api/model/PaymentEvent.java]: connector event doesn't have a chargeId

File: src/test/java/uk/gov/pay/api/service/GetPaymentEventServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        PaymentEventsResponse paymentEventsResponse = getPaymentEventService.getPaymentEvent(account, "abc123");
+        assertThat(paymentEventsResponse.getPaymentId(), is("abc123"));
+        assertThat(paymentEventsResponse.getLinks().getSelf().getHref(), is("http://publicapi.test.localhost/v1/payments/abc123/events"));
+        assertThat(paymentEventsResponse.getEvents().size(), is(2));
+        assertThat(paymentEventsResponse.getEvents().get(0).getPaymentId(), is("abc123"));
+        assertThat(paymentEventsResponse.getEvents().get(0).getPaymentLink().getPaymentLink().getHref(), is("http://publicapi.test.localhost/v1/payments/abc123"));
+        assertThat(paymentEventsResponse.getEvents().get(0).getState().getStatus(), is("created"));
+        assertThat(paymentEventsResponse.getEvents().get(0).getState().isFinished(), is(false));
+        assertThat(paymentEventsResponse.getEvents().get(1).getPaymentId(), is("abc123"));
+        assertThat(paymentEventsResponse.getEvents().get(1).getPaymentLink().getPaymentLink().getHref(), is("http://publicapi.test.localhost/v1/payments/abc123"));
```
Comment (jankowiakmaria) [inline src/test/java/uk/gov/pay/api/service/GetPaymentEventServiceTest.java]: would be good to check whether `updated` is populated as well

File: src/test/resources/pacts/publicapi-connector-get-payment-events.json [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+              ]
+            },
+            "$.events[1].state.finished": {
+              "matchers": [
+                {
+                  "match": "type"
+                }
+              ]
+            },
+            ,
```
[the remark this answers is not in the document]
Comment (jankowiakmaria) [reply src/test/resources/pacts/publicapi-connector-get-payment-events.json]: addressed in https://github.com/alphagov/pay-publicapi/pull/633/commits/14311e18b8df8018761b8be7218dbe73e6332284

File: src/main/java/uk/gov/pay/api/model/PaymentEvent.java (hunk 1 above)
Comment (jankowiakmaria) [reply src/main/java/uk/gov/pay/api/model/PaymentEvent.java]: addressed in https://github.com/alphagov/pay-publicapi/pull/633/commits/14311e18b8df8018761b8be7218dbe73e6332284

File: src/test/java/uk/gov/pay/api/service/GetPaymentEventServiceTest.java (hunk 2 above)
Comment (jankowiakmaria) [reply src/test/java/uk/gov/pay/api/service/GetPaymentEventServiceTest.java]: addressed in https://github.com/alphagov/pay-publicapi/pull/633/commits/14311e18b8df8018761b8be7218dbe73e6332284

File: src/main/java/uk/gov/pay/api/model/PaymentEvents.java [hunk 4]
```diff
@@ -1,50 +1,22 @@
 package uk.gov.pay.api.model;
 
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.fasterxml.jackson.databind.JsonNode;
-import io.swagger.annotations.ApiModel;
-import io.swagger.annotations.ApiModelProperty;
 import uk.gov.pay.api.model.links.PaymentLinksForEvents;
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/api/model/PaymentEvents.java]: Unused import

File: src/main/java/uk/gov/pay/api/model/PaymentEventsResponse.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+@ApiModel(value="PaymentEvents", description = "A List of Payment Events information")
+public class PaymentEventsResponse {
+    @JsonProperty("payment_id")
+    private final String paymentId;
+
+    private final List<PaymentEventResponse> events;
+
+    @JsonProperty("_links")
+    private PaymentLinksForEvents links = new PaymentLinksForEvents();
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/api/model/PaymentEventsResponse.java]: Do we initialiser here ? Looks like it is always set in constructor

File: src/main/java/uk/gov/pay/api/service/GetPaymentEventsService.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+import javax.inject.Inject;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.net.URI;
+
+import static org.apache.http.HttpStatus.SC_OK;
+
+public class GetPaymentEventsService {
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/api/service/GetPaymentEventsService.java]: Can we not use `ConnectorService` to get Payment events from connector. I feel ConnectorService is set for this purpose  or I am happy if the plan is to use ConnectorService eventually

Comment (jankowiakmaria) [reply src/main/java/uk/gov/pay/api/service/GetPaymentEventsService.java]: there will be further refactoring to use `ConnectorService`

File: src/main/java/uk/gov/pay/api/model/PaymentEventsResponse.java (hunk 5 above)
Comment (jankowiakmaria) [reply src/main/java/uk/gov/pay/api/model/PaymentEventsResponse.java]: Can I remove it in follow-up PR?

File: src/main/java/uk/gov/pay/api/model/PaymentEvents.java (hunk 4 above)
Comment (jankowiakmaria) [reply src/main/java/uk/gov/pay/api/model/PaymentEvents.java]: Can I remove it in follow-up PR?

**Pull request merged**